### PR TITLE
Flip32 F4 target

### PR DESCRIPTION
**Pull request by DzikuVx**
Flip32 F4 / Airbot F4 target. This is Revo target with few changes.

Depends on updated Revo target ( #688 )
- [x] verify I2C working
- [x] add all supported I2C hardware
- [x] PPM
- [x] SerialRx
- [x] BMP085
- [x] BMP280
- [x] MS5611
- [x] HMC5883
- [x] verify GPS
- [x] Buzzer on dedicated pins
- [x] Voltage monitoring
- [x] Current monitoring
- [x] RSSI ADC
- [ ] verify LED working, move to dedicated pins if possible
- [x] servos
- [x] disable Parallel PWM

### CODE REVIEW

**Comment by nathantsoi** [comment]
note, this is also the same as the omnibusf4, which adds the max chip, when osd support is added to inav

**Comment by nathantsoi** [comment]
sweet, nice work! note that this is also the same as the omnibusf4, which adds the max chip, when osd support is added to inav. and an fyi, we decided in betaflight to make the airbot f4 / flip 32 work w/ the revo target.

**Comment by DzikuVx** [comment]
@nathantsoi We need I2C for Baro/Mag/WhateverElse and current implementation does not work for both original Revo and Flip32F4. 
See #718 an #719 
With #718 it would be possible to merge revo and flip32f4 again

The same problem goes for Betaflight as well

**Comment by DzikuVx** [comment]
@nathantsoi Update: because build in Revo mag and baro uses I2CDEV_1 and Flip32F4 and derivatives does not have built in baro and mag and only I2CDEV_2 is available it would be impossible to use them in Revo target.
Even when 2 buses will be implemented, board would try to use I2CDEV_1 for them even when they are on I2CDEV_2
So new target is requires after all

File: src/main/main.c [hunk 1]
```diff
@@ -400,7 +400,7 @@ void init(void)
             i2cInit(I2C_DEVICE);
         }
     }
-#elif defined(CC3D)
+#elif defined(I2C_DEVICE_SHARES_UART3)
```
Comment (DzikuVx) [inline src/main/main.c]: @digitalentity @martinbudden that's my idea of removing some target dependencies in main.c

Comment (digitalentity) [reply src/main/main.c]: I think we need to be smarter - let the per-target code decide what is available.

**Comment by DzikuVx** [comment]
Either REVO target is wrong, or this board is not REVO clone after all. Motor 6 is not connected to PA0 like target file states, but PA8. That is actually good, since it allows RSSI ADC and Motor 6 to be used at the same time.

LED strip not working, reason unknown

**Comment by martinbudden** [comment]
@DzikuVx , have you tried LED strip with blackbox turned off. In betaflight there have been some issues of DMA clashing (both LED strip and blackbox use DMA), so a similar thing might be occurring here.

Also, stating the obvious, I presume you have also checked the pin for motor 5?

**Comment by DzikuVx** [comment]
No, did not tried without Blackbox enabled. Will check that today.

Yes, I checked pin for motor 5, checked timer clashes and all that stuff. Since I do not see it as primary function, it can be published even with LED disabled

**Comment by martinbudden** [comment]
I agree, we can add this target without LED strip, as long as it's clearly stated in the release notes.

It would be good to find out what is going on though.

**Comment by DzikuVx** [comment]
I suspect led driver on our side. In bf 3.0.1 it is working, but have to
recheck 3.1

wt., 25.10.2016, 10:12 użytkownik Martin Budden notifications@github.com
napisał:

**Comment by DzikuVx** [comment]
@digitalentity ready for merge into 1.4
LED Strip not working...

**Comment by DzikuVx** [comment]
Flight tested with async gyro.... I love it :)

File: src/main/target/FLIP32F4/target.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#define I2C_DEVICE_SHARES_UART3
+
+#define USE_ADC
+#define CURRENT_METER_ADC_PIN   PC1
+#define VBAT_ADC_PIN            PC2
+#define RSSI_ADC_PIN            PA0
+
+#define SENSORS_SET (SENSOR_ACC|SENSOR_MAG|SENSOR_BARO)
+
+#define LED_STRIP
```
Comment (digitalentity) [inline src/main/target/FLIP32F4/target.h]: Since LED_STRIP is not working should't we disable it?

**Review by digitalentity — CHANGES_REQUESTED** [review]

File: src/main/target/FLIP32F4/target.h (hunk 2 above)
Comment (DzikuVx) [reply src/main/target/FLIP32F4/target.h]: Yes, you are right. I will disable it

Comment (DzikuVx) [reply src/main/target/FLIP32F4/target.h]: @digitalentity  and done

**Review by digitalentity — APPROVED** [review]

**Comment by DzikuVx** [comment]
OK, it will not be any better than is right now, I'm merging it

**Pull request merged**